File: leetcode/easy/reverse-integer.py
def reverse(x):
    rev = 0
    sign = -1 if x < 0 else 1

    x = abs(x)
    while(x > 0):
        lastd = x % 10
        x = int(x / 10)
        rev = (rev * 10) + lastd

    rev = rev * sign
    if rev > 0x7fffffff or rev < -0x80000000:
        return 0
    return rev

print(reverse(-9463847412))

File: leetcode/easy/reverse-integer.js

var reverse = function(x) {
    let rev = 0;

    let sign = x > 0 ? 1 : -1;

    x = Math.abs(x);
    while(x != 0){
        let digit = x % 10 ;
        rev = rev * 10 + digit;
        x = Math.floor(x / 10);
    }

    if (rev > 0x7fffffff || rev < -0x80000000) {
        return 0;
    }

    rev = sign * rev;
    return rev;
};

console.log(reverse(-123));
